algorithm/pge-eda.metta

```diff
@@ -0,0 +1,564 @@
+;; module import
+
+; ! (register-module! ../../metta-moses)
+
+; ! (import! &self metta-moses:utilities:general-helpers)
+; ! (import! &self metta-moses:scoring:fitness)
+
+;; python bindings for random number generation and rounding
+
+! (bind! np (py-atom numpy))
+! (bind! round (py-atom round))
+
+;; space bindings
+
+! (bind! &grammar (new-space))
+! (bind! &geno (new-space))
+! (bind! &count (new-space))
+! (bind! &population (new-space))
+! (bind! &scored (new-space))
+
+;; adding the grammar
+
+! (add-reduct &grammar 
+                (superpose 
+                        ((start expr 0)
+                         (expr (bop expr expr) 0)
+                         (expr (uop expr) 1)
+                         (expr term 2)
+                         (bop AND 0)
+                         (bop OR 1)
+                         (uop NOT 0)
+                         (term A 0)
+                         (term B 1))))
+
+;; global constants -- 
+;;                length of grammar (L), 
+;;                set of grammar non-terminals (NTS) , 
+;;                test data (DATA)  , 
+;;                max-wrap (WRAP), 
+;;                round-off precision (PREC)
+;;                Generation count (G)
+;;                Expression count (E)
+
+! (bind! L (sum` (collapse (match &grammar ($nt $r $i) 1))))
+! (bind! NTS (collapse (unique (match &grammar ($nt $r $i) $nt))))
+
+! (bind! DATA ((False (A False) (B False))
+                (False (A False) (B True))
+                (False (A True) (B False))
+                (True (A True) (B True))))
+
+! (bind! WRAP 1)
+! (bind! MAXITER (* L WRAP))
+! (bind! PENALITY 0.1) 
+! (bind! FACTOR 0.05)
+! (bind! PREC 3)
+! (bind! G 2)
+! (bind! E 1)
+
+;; ::::::::::::: CREATING GENOTYPE :::::::::::::
+;; createGenotype -- creates genptype of a specified length in the range taking random values in the range [0,1] from the uniform distribution
+;;                   makes use of python/numpy uniform distributed random variable generator
+;;              $n -- length of the genotype tuple
+;;              $prec -- rounding off precision value
+;;              $random -- random number to prevent metta from caching results
+;;              Examples :- 
+;;                      (createGenotype 9 3 dummy)
+;;                      -- 1st run -- (0.227 0.584 0.498 0.364 0.912 0.881 0.504 0.206 0.119)
+;;                      -- 2nd run -- (0.71 0.511 0.312 0.676 0.281 0.669 0.128 0.857 0.848)
+;;                      -- 3rd run -- (0.467 0.206 0.563 0.957 0.029 0.907 0.423 0.675 0.231)
+
+(: (createGenotype (-> Number Number Expression)))
```

Comment:
The type and definition of this function isn't the same. I see the function takes three arguments but the type says it takes two. If it doesn't introduce any bugs, try updating the type to match the definiton.
```suggestion
(: (createGenotype (-> Number Number Number Expression)))
```